user-ssh-builders/README.md

```diff
@@ -0,0 +1,174 @@
+  instance is already running. We don't want to kill it and spawn a
+  new one. We probably need to print it was not possible to spawn a
+  new instance into the builder-live/backend log
+- Where to print the instructions on how to connect to the builder?
+- If the build fails in multiple chroots, which one of those builders
+  should be kept alive? We can't keep them all.
+- Inability to select an instance type in advance so it may be hard
+  to debug builds where e.g. Amazon AWS Spot instance is needed
+  and nothing else.
+
+### Option 3 - Resubmit and keep for ssh
+
+Next to the "Resubmit" button, we could have another "Resubmit and
+keep for ssh" button. This will be the case for both failed and
+succeeded builds. It is not guaranteed that user will get a builder
+from the same pool but this might turn out not to be a problem, we
+will see. It should be easy to improve this if needed - we could start
+tracking builder pool for each build and then use this information
+when resubmitting.
+
+This option will be the most user friendly. No settings, no
+architecture/pool selection, etc.
+
+
+### Common ground
+
+Since the workflow starts from a failed build, we could create a MOTD
+on the builder showing instructions on how to reproduce the particular
+build using `copr-rpmbuild`.
+
+
+## Instance type
+
+This is probably relevant only for `Workflow Option 1 - Button`.
+
+It is not enough to give acess to a random builder. Users needs to be
+able to specify at least an architecture. But that is IMHO not good
+enough. For example we have x86_64 builders our own hypervisors and in
+Amazon AWS. Historically, we had issues that appeared only on builders
+from one of them.
+
+Even though it architectonically doesn't make any sense for Copr to
+know about resalloc pools, I think we should let users select a resalloc
+pool from which to spawn the builder. We can point them to the
+[Resalloc Pools][resalloc-pools] page for details.
+
+We have the resalloc `pools.yaml` in ansible, so it shouldn't be hard
+to query only their names from the config and store them into some
+config file on copr-frontend.
+
+
+## Auth
+
+We decided to go with `Option 1 - SSH key`.
+
+### Option 1 - SSH key
+
+There is a [FAS API][fas-api] and it can query each user and their
+SSH key, e.g. [mine is here][fas-api-frostyx]. But it requires auth,
+not sure if `fkinit` or recently logging into some Fedora service,
+but try it in an anonymous window, it doesn't work.
+
+Fetching SSH keys without authentication
+[requested here][keys-without-auth].
+
+Anyway, Copr already fetches some user information from FAS so I think
+we can assume we will be able to fetch the SSH keys as well.
+
+For other Copr instances, we will need to fetch SSH keys from LDAP,
+etc.
+
+Alternativelly, we could allow users to specify a public URL to their
+SSH key, upload it, or copy-paste it to some Copr form, but I don't
+like any of these options.
```

Comment:
Being able to set custom SSH key(s) would allow Packit to provide this to its users. We can easily let either people configure the public key or take it from GitHub. The workflow I see here:


1. Someone opens a PR.
2. Build is triggered.
3. Build fails, in the results (or as a comment), Packit suggests the user to try this feature.
4. User triggers this (e.g. via `/packit rerun-with-ssh` or some other comment)
5. Packit uses the Copr Python client to use this feature and set the user's public SSH key (it should be available via GitHub API). 
7. Packit adds a comment with the instructions on how to connect.